feat: camel k 1.12.0 release (#981)
apache/camel-website: content/blog/2023/02/camel-k-release-1-12/index.md

---
title: "Camel K 1.12 release"
date: 2023-02-27
draft: false
authors: [squakez]
categories: ["Releases", "Camel K"]
preview: "What's new in Camel K 1.12"
---

Image (c): <a href="https://www.vecteezy.com/free-vector/camel">Camel Vectors by Vecteezy</a>

> Altough we're already in [Camel K 2.0](/blog/2023/02/camel-k-roadmap-2023/) mode, we had not forgotten to bring new fresh features on version 1. **Camel K 1.12.0** has been just released and it brings a lot of new interesting upgrades and features. This may be the last release of Camel K 1 major version as we aim to have a release of Camel K 2 during half of this year.

Reviewer: ```suggestion
Although we're already in [Camel K 2.0](/blog/2023/02/camel-k-roadmap-2023/) mode, we had not forgotten to bring new fresh features on version 1. **Camel K 1.12.0** has just been released and it brings a lot of new interesting upgrades and features. This may be the last release of Camel K 1 major version as we aim to have a release of Camel K 2 during half of this year.
```

As usual let's start by acknowledging the tech stack on top of Camel K and the new exciting versions our fellows have baked:

* Apache Camel K Runtime 1.17.0
* Apache Camel Quarkus 2.16.0
* Apache Camel 3.20.1
* Apache Camel Kamelets 3.20.1.1

Thanks to Apache Camel, Camel Quarkus and Kamelet Catalog contributors for the great efforts they've put in those new releases. You may notice a slight change compared to the previous releases. It's about Kamelets versioning which is now aligned to the framework version (yes, it will be easier to remember from now on!).

## Java, Groovy and Kotlin Quarkus native support

The most interesting feature coming in this release is the extension of **Quarkus Native** mode to Java, Groovy and Kotlin DSLs. This feature was available only on YAML and XML so far, but since version 1.12.0 you may enjoy the capability of resource shrinking also on other DSLs. Just [mark your Integration as Quarkus Native](/camel-k/next/traits/quarkus.html) and let the operator to do the heavy lift.

## `kamel promote` dry run

> The `kamel promote` command has simplified the promotion of an Integration to higher environment. `kamel promote -o yaml` is doing the very same stuff, but, instead of publishing the Integration in the new namespace, it returns the yaml (or json if you prefer) specification. It's a little step towards GitOps and will allow you perform already some integration with external pipeline tools.

Reviewer: ```suggestion
The `kamel promote` command has simplified the promotion of an Integration to a higher environment. `kamel promote -o yaml` is doing the very same stuff, but, instead of publishing the Integration in the new namespace, it returns the yaml (or JSON if you prefer) specification. It's a little step towards GitOps and will allow you to perform already some integration with external pipeline tools.
```

## Generate apply configurations

> This is something not immediatly user facing, as it enables API types to be used with server-side apply. Still worth mentioning to let Camel K developers around the world this is now available.

Reviewer: ```suggestion
This is something not immediately user-facing, as it enables API types to be used with server-side apply. Still worth mentioning to let know Camel K developers around the world this is now available.
```
